specification/profiling.md

```diff
@@ -0,0 +1,169 @@
+# Profiling Conventions
+
+**Status**: [experimental](../README.md#versioning-and-status-of-the-specification)
+
+This document describes the specific behavior and conventions for Splunk
+agents that contain profiling features.
+
+- [Profiling Conventions](#profiling-conventions)
+    * [Activation](#activation)
+    * [Call Stack Sampling](#call-stack-sampling)
+        + [Call Stack Filtering](#call-stack-filtering)
+        + [Call Stack Span Association](#call-stack-span-association)
+        + [Call Stack Ingest](#call-stack-ingest)
+    * [ResourceLogs](#resourcelogs)
+        - [InstrumentationLibraryLogs](#instrumentationlibrarylogs)
+        - [LogRecord Common Attributes](#logrecord-common-attributes)
+        - [LogRecord Fields](#logrecord-fields)
+            * [Call Stack Format](#call-stack-format)
+- [References](#references)
+
+## Activation
+
+Agents may choose to activate or deactivate their profiler by default.
+A configuration setting MUST be provided to allow overriding the default: 
+
+* `SPLUNK_PROFILER_ENABLED` environment variable
+
+The default setting for `SPLUNK_PROFILER_ENABLED` MUST be clearly documented. 
+The agent SHOULD NOT allow changing the setting at runtime, and the initial 
+setting SHOULD be used for the entire lifespan of the application run.
+
+An agent whose profiling capability is deactivated MUST NOT introduce
+additional profiling-based overhead. It also MUST NOT ingest profiling-based
+data.
+
+## Call Stack Sampling
+
+An agent that has profiling capabilities MUST be able to sample call stacks
+at a fixed interval. This default interval MUST default to 1 second.
+
+If a language agent runtime supports threading, stacks MUST be sampled across
+all process threads. The samples for all threads SHOULD be taken instantaneously
+and, in the event that this is not feasible, MUST be taken as close together
+as possible. If the samples are taken consecutively, then the profiler MUST 
+use a consistent ordering strategy (such as thread name or ID) when sampling all
+threads.
+
+### Call Stack Filtering
+
+Various runtimes MAY contain internal and other threads that are undesirable
+to include for profiling. This could include threads that are internal to 
+runtime behavior or agent internal workings. The choice of which threads are 
+undesirable is implementation specific and not defined.
```

Comment:
I am not sure here if it is a good recommendation to exclude any threads. Especiially I am concerned about:

> This could include threads that are internal to runtime behavior or agent internal workings. 

E.g. in .NET, the profiles of GC threads are extremely valuable to detect problems with memory allocation. Without the "internal working" profiles we may have trouble finding issues connected with the internals.

Reply:
Fair enough, that's why it's a SHOULD. In the dotnet case, if those threads are crucial to memory profiling and you need to keep them, so be it. I think that would classify as a desirable thread then and still meet the spec.